### PR TITLE
PRO-2191 download

### PR DESCRIPTION
**Pull request by abea**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -10,11 +10,15 @@ module.exports = {
           label: 'Export',
           route: '/export',
           messages: {
-            progress: 'Exporting complete.',
-            completed: 'Exporting {{ type }}...'
```
Comment (abea) [inline index.js:14]: We're going to trigger this directly to include the event info.

File: lib/export.js [hunk 2]
```diff
@@ -126,8 +126,26 @@ module.exports = (self) => {
               return reject(error);
             }
 
+            const downloadUrl = self.apos.attachment.uploadfs.getUrl() + downloadPath;
```
Comment (boutell) [inline lib/export.js:129]: Do you really get a download this way across browsers with just `_blank`? I would think you would need to use a route that sends with with `Content-Disposition: Attachment` (I think I sent you the 2.x route we use for it).

**Review by boutell — CHANGES_REQUESTED** [review]
We might need content-disposition here, let me know if I'm wrong. Maybe we don't simply because none of the output formats are native to the browser anyway, but it's a good idea to specify downloading is the intent.

File: lib/export.js (hunk 2 above)
Comment (abea) [reply lib/export.js:129]: The only one I see that does not download is Safari, which opens the CSV as text in a new tab. This is essentially the way we handled it in the A2 version. https://github.com/apostrophecms/apostrophe-pieces-export/blob/main/public/js/export-modal.js#L39

Comment (abea) [reply lib/export.js:129]: Since we're saving the file to the public directory I'm not sure how much we get out of putting it behind another route.

**Review by boutell — CHANGES_REQUESTED** [review]
In the A2 version we used a route to download it and that route emits the right Content-Disposition header, as a result of which I betcha Safari doesn't show it in a tab.

**Comment by boutell** [comment]
(In A3 this would need to be in `routes` because we need to respond directly via `res` and in A3 `apiRoutes` just return a body.)

```
  // Provides a simple route to download a file as an attachment, rather than
  // viewing it. Pipes it from the regular public URL to avoid acting as a
  // bypass of permissions
  self.apiRoute('get', 'download', async (req, res) => {
    const _id = self.apos.launder.id(req.query._id);
    const info = await self.db.findOne({
      _id: _id
    });
    if (!info) {
      return res.status(404).send('notfound');
    }
    const url = self.url(info, { size: 'original' });
    res.setHeader('Content-Disposition', `attachment; filename="${path.basename(url)}"`);
    res.setHeader('Content-Type', 'application/octet-stream');
    const resolvedUrl = (new URL(url, req.absoluteUrl)).toString();
    await request(resolvedUrl).pipe(res);
  });
```

**Comment by boutell** [comment]
In A3 that route is in core, as there can be other applications for reliably downloading rather than viewing an attachment, which I think is still the case.

**Comment by abea** [comment]
> This is essentially the way we handled it in the A2 version.
By that I meant it's how it was done in the pieces export module.

**Review by boutell — APPROVED** [review]

**Pull request merged**